code/__HELPERS/game.dm

```diff
@@ -152,11 +152,11 @@
 /proc/recursive_hear_check(O)
 	var/list/processing_list = list(O)
 	. = list()
-	while(processing_list.len)
-		var/atom/A = processing_list[1]
+	var/i = 1
+	while(i < length(processing_list))
```

Comment:
```suggestion
	while(i <= length(processing_list))
```
Otherwise change `i` value to 0 and `i++` to `++i`

Reply:
done, i++ is faster anyways